distributed/client.py

```diff
@@ -394,6 +396,16 @@ def __init__(self, address=None, start=True, loop=None, timeout=5,
         self.connection_args = self.security.get_connection_args('client')
         self._connecting_to_scheduler = False
 
+        if loop is None:
+            self._should_close_loop = None
+            if in_ioloop():
+                self.loop = IOLoop.current()
```

Comment:
I agree, however there are some odd cases where this changes behavior in ways that I don't totally understand.  In particular when starting many clients on the same IOLoop running in a separate thread such as often occurs when using `worker_client`.